### PR TITLE
Bug #211961 Photo attendance: As informed to us, any two days will be photo mandatory attendance not more than that. Photo attendance is coming for 3 to 4 days on PMS

### PR DESCRIPTION
**Pull request by Tusharmahajan12**
https://tracker.tekdi.net/issues/211961
# I have ensured that following `Pull Request Checklist` is taken care of before sending this PR

* [x] Code is formatted as per format decided
* [x] Updated acceptance criteria in tracker
* [x] Updated test cases in test-cases-tracker
* [x] Updated new API endpoint if any in common postman collection
* [x] Updated DB changes in db-tracker if any

## Summary by CodeRabbit

- **Refactor**
	- Enhanced the attendance data processing for improved reliability and accuracy.

### CODE REVIEW

File: src/src/camp/camp.service.ts [hunk 1]
```diff
@@ -3646,87 +3646,69 @@ export class CampService {
...
+		AND a.photo_1 IS NOT NULL AND a.photo_1 != '-'
+		AND cda.camp_id = ${camp_id}
+		AND a.context = 'camp_days_activities_tracker'
+	  GROUP BY gs.date
+	  ORDER BY gs.date;
 		;`;
+
 		const attendance_data = (
 			await this.hasuraServiceFromServices.executeRawSql(sql)
 		)?.result;
```
Comment (sagarkoshti1990) [inline src/src/camp/camp.service.ts:3675]: Add try catch and in catch return query error.

File: src/src/camp/camp.service.ts [hunk 2]
```diff
@@ -3646,87 +3646,69 @@ export class CampService {
...
-
-		// Calculate the difference in days
-		const today = new Date();
-		const differenceInTime = Math.abs(
-			today.getTime() - highestDate.getTime(),
-		);
-		const differenceInDays = Math.floor(
-			differenceInTime / (1000 * 3600 * 24),
+		const data = await this.hasuraServiceFromServices.getFormattedData(
+			attendance_data,
```
Comment (sagarkoshti1990) [inline src/src/camp/camp.service.ts:3678]: check attendance_data is in correct format like array, Object or string .

File: src/src/camp/camp.service.ts [hunk 3]
```diff
@@ -3646,87 +3646,69 @@ export class CampService {
...
 		);
 
-		if (filteredDates.length == 0) {
+		if (data.length == 0) {
```
Comment (sagarkoshti1990) [inline src/src/camp/camp.service.ts:3681]: check here data is in array format.

File: src/src/camp/camp.service.ts [hunk 4]
```diff
@@ -3646,87 +3646,69 @@ export class CampService {
...
 			return res.json({
 				learner_camp_attendance_data: 1,
+				mesage: 'data not found',
 			});
-		}
+		} else if (data.length == 1) {
```
Comment (sagarkoshti1990) [inline src/src/camp/camp.service.ts:3686]: check here data is in array format.

**Review by sagarkoshti1990 — CHANGES_REQUESTED** [review]

File: src/src/camp/camp.service.ts (hunk 1 above)
Comment (Tusharmahajan12) [reply src/src/camp/camp.service.ts:3675]: resolved

File: src/src/camp/camp.service.ts (hunk 2 above)
Comment (Tusharmahajan12) [reply src/src/camp/camp.service.ts:3684]: resolved

File: src/src/camp/camp.service.ts (hunk 3 above)
Comment (Tusharmahajan12) [reply src/src/camp/camp.service.ts]: resolved

File: src/src/camp/camp.service.ts (hunk 4 above)
Comment (Tusharmahajan12) [reply src/src/camp/camp.service.ts]: resolved

**Review by sagarkoshti1990 — APPROVED** [review]

**Pull request merged**